corporate/lib/stripe.py

```diff
@@ -4255,14 +4272,59 @@ def get_plan_renewal_or_end_date(plan: CustomerPlan, event_time: datetime) -> da
     return billing_period_end
 
 
+def insufficient_data_to_generate_invoice(server: RemoteZulipServer, event_time: datetime) -> bool:
+    assert server.last_audit_log_update is not None
+
+    # 'last_audit_log_update' is updated hourly.
+    return event_time - server.last_audit_log_update >= timedelta(days=1)
```

Comment:
I think the right algorithm is that:
* We should never invoice a plan where `server.last_audit_log_update < next_invoice_date` -- but we don't need to throw an exception if that temporal difference is small.
* If that temporal difference is greater than a day, we should send the email to our support team to investigate.
* We probably need some way to track the last date for which we sent such an email -- we don't want to generate such an email every hour/day.

Reply:
Oh, I think there was a miscommunication. The issue states:

> Probably we want to complain in a way that the Zulip team will see it whenever we are due for invoicing and that is overdue by more than a day.

**whenever we are due for invoicing and that is overdue by more than a day.**

I thought it as "one day has passed since the date (next_invoice_date) invoicing should occur"

Reply:
> We probably need some way to track the last date for which we sent such an email

Added the following field in CustomerPlan:
```python
    # Flag to track if an email has been sent to Zulip team for
    # invoice overdue by >= one day. Helps to send an email only once
    # and not every time when cron run.
    invoice_overdue_email_sent = models.BooleanField(default=False)
```
Also, verified the behaviour in tests.